internal/provisioner/helm_utils.go

```diff
@@ -148,6 +148,9 @@ func upgradeHelmChart(chart helmDeployment, configPath string, logger log.FieldL
 	if err != nil {
 		return errors.Wrap(err, "failed to get values file")
 	}
+	if strings.HasPrefix(chart.desiredVersion.ValuesPath, os.TempDir()) {
+		defer os.Remove(chart.desiredVersion.ValuesPath)
+	}
```

Comment:
Instead of this check, you could modify `fetchFromGitlabIfNecessary` to return a cleanup function as well and then defer that. May be a less brittle, but this should work fine regardless.

Reply:
Sure! Oh I like that idea a lot actually thanks @gabrieljackson